### PR TITLE
Add support for Mixed query aggregations

### PR DESCRIPTION
**Pull request by clementetb**
This PR adds support for query aggregation operations on Mixed field types.

⚠️ Do not merge there is still an open issue on the min aggregation function:
https://github.com/realm/realm-core/issues/4571

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 1]
```diff
@@ -1946,7 +1947,7 @@ public boolean isValid() {
     /**
      * Calculates the sum of a given field.
```
Comment (rorbech) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java:1948]: Sounds like the doc could use some comments on what the actual values are when used on mixed types.

**Review by rorbech — COMMENTED** [review]
LGTM, except that I think we need some more documentation of the semantics on the individual aggregations. Maybe I have missed it because of not having review the other mixed PRs, but in that case we at least need some references.

Also some minor comments on the test for min w/wo nullable elements

File: realm/realm-library/src/androidTest/kotlin/io/realm/mixed/MixedQueryTests.kt [hunk 2]
```diff
@@ -102,12 +106,54 @@ class MixedQueryTests {
...
+        initializeTestData()
+        val value = realm.where<MixedNotIndexed>().sum(MixedNotIndexed.FIELD_MIXED)
+        assertEquals(Decimal128.parse("279.0"), value)
+    }
+
+    @Test
+    fun min() {
+        initializeTestData()
+        val value = realm.where<MixedNotIndexed>().minMixed(MixedNotIndexed.FIELD_MIXED)
+        assertNotNull(value)
```
[the remark this answers is not in the document]
Comment (clementetb) [reply realm/realm-library/src/androidTest/kotlin/io/realm/mixed/MixedQueryTests.kt:132]: I am waiting for the core to resolve the issue to set it to the right value from my understanding it shall be `false`.

File: realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp [hunk 3]
```diff
@@ -233,6 +247,27 @@ JNIEXPORT jobject JNICALL Java_io_realm_internal_TableQuery_nativeMaximumDouble(
...
+                                                                               jlong columnKey) {
+    Query *pQuery = Q(nativeQueryPtr);
+    ConstTableRef pTable = pQuery->get_table();
+    if (!TYPE_VALID(env, pTable, columnKey, col_type_Mixed)) {
+        return nullptr;
+    }
+    try {
+        ObjKey return_ndx;
+        Mixed result = pQuery->maximum_mixed(ColKey(columnKey), &return_ndx);
+        if (bool(return_ndx)) {
```
[the remark this answers is not in the document]
Comment (clementetb) [reply realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp:261]: 🤦🏻‍♂️

File: realm/realm-library/src/androidTest/kotlin/io/realm/mixed/MixedQueryTests.kt [hunk 4]
```diff
@@ -109,12 +111,75 @@ class MixedQueryTests {
...
+    fun sum() {
+        initializeTestData()
+        val value = realm.where<MixedNotIndexed>().sum(MixedNotIndexed.FIELD_MIXED)
+        assertEquals(Decimal128.parse("279.0"), value)
+    }
+
+    // This test case is meant to catch when  https://github.com/realm/realm-core/issues/4571 gets fixed
+    @Test
+    fun catch_minAggregationFixed() {
+        initializeTestData(true)
```
Comment (nhachicha) [inline realm/realm-library/src/androidTest/kotlin/io/realm/mixed/MixedQueryTests.kt:136]: can we use named argument to increase readbility?

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 5]
```diff
@@ -1946,10 +1947,10 @@ public boolean isValid() {
...
      *
-     * @param fieldName the field to sum. Only number fields are supported.
+     * @param fieldName the field to sum. Only number and Mixed fields are supported.
      * @return the sum of fields of the matching objects. If no objects exist or they all have {@code null} as the value
      * for the given field, {@code 0} will be returned. When computing the sum, objects with {@code null} values
-     * are ignored.
+     * are ignored. When applied on a Mixed field it would returning type would be of type Decimal128.
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java:1953]:
```suggestion
     * are ignored. When applied to a Mixed field, the returning type will be {@code Decimal128}.
```
?

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 6]
```diff
@@ -2021,6 +2024,25 @@ Decimal128 averageDecimal128(String fieldName) {
         return query.averageDecimal128(columnIndex);
     }
 
+    /**
+     * Returns the average of a given field.
+     * Does not support dotted field notation.
+     *
+     * @param fieldName the field to calculate average on. Only Mixed fields is supported. For other types consider using {@link #average(String)}.
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java:2031]:
```suggestion
     * @param fieldName the field to calculate average on. Only Mixed fields are supported. For other types consider using {@link #average(String)}.
```

**Review by nhachicha — COMMENTED** [review]
LGTM

**Pull request merged**